sycl/source/detail/queue_impl.hpp

```diff
@@ -49,6 +49,18 @@ enum QueueOrder { Ordered, OOO };
 
 class queue_impl {
 public:
+  // \return a default context for the platform if it includes the device
+  // passed, a new context otherwise.
+  static ContextImplPtr getDefaultOrNew(const DeviceImplPtr &Device) {
```

Comment:
We cannot use the default context(which is bound to root devices) to create a queue for a sub-device, so here we create a new context if for some reason the default context doesn't include target device.
We could bound all sub-devices to the same default context as well, but it might degrade performance.
Also, there are a lot of ways to create sub-devices from a root device, so it's not clear which sub-devices should be included.

Reply:
Tagging @jbrodman